Makefile

```diff
@@ -45,14 +45,17 @@ doc: ## Generate and check documentation
 .PHONY: test-default
 test-default: ## Run tests with default features
 	$(DEBUG_OVERFLOW_INFO) cargo nextest run --profile default --release --all-features
+	$(DEBUG_OVERFLOW_INFO) cargo test --doc --release --all-features
 
 .PHONY: test-smt-hashmaps
 test-smt-hashmaps: ## Run tests with `smt_hashmaps` feature enabled
 	$(DEBUG_OVERFLOW_INFO) cargo nextest run --profile default --release --features smt_hashmaps
+	$(DEBUG_OVERFLOW_INFO) cargo test --doc --release --features smt_hashmaps
 
 .PHONY: test-no-std
 test-no-std: ## Run tests with `no-default-features` (std)
 	$(DEBUG_OVERFLOW_INFO) cargo nextest run --profile default --release --no-default-features
+	$(DEBUG_OVERFLOW_INFO) cargo test --doc --release --no-default-features
```

Comment:
I think it's better to separate these out into a dedicated make recipe and introduce a dedicated CI job for this like in miden-base:

https://github.com/0xMiden/miden-base/blob/291d6620bc157e81f76c37b3dbcfb64ac5f80d8c/.github/workflows/test.yml#L33-L44

https://github.com/0xMiden/miden-base/blob/291d6620bc157e81f76c37b3dbcfb64ac5f80d8c/Makefile#L81

That way, when using those make commands locally, it's a bit quicker for a test run to finish (while obviously testing less). It should also speed up the CI because we can build and run doc tests in parallel.

Btw. is `--release` mode necessary for doc tests? Using that will result in much longer test build times.

Reply:
Ah, good point. I think I would still prefer the miden-base setup but not a strong opinion at all.